### PR TITLE
Remove i-PI distribution from tools folder

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

As discussed in PR #4157, the i-PI distribution is removed from LAMMPS and replaced with instructions to install it via pip.

**Related Issue(s)**

PR #4157 

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

This places a README.md file in the tools folder, but also updates the doc/src/Tools.rst file and the documentation for fix ipi accordingly.

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system

### CODE REVIEW

**Comment by ceriottm** [comment]
I'm waiting for #4157 to be merged as I also touched the docs in there - then I'll take care of merging the changes in here. Thanks for helping, the paper is now being revised so it's nice that the LAMMPS driver will also be updated in time for v3.0!

**Comment by akohlmey** [comment]
> I'm waiting for #4157 to be merged as I also touched the docs in there

@ceriottm No problem. I figured it would simplify the process, if I provide a template with all the changes included that we usually apply when removing some feature.

**Review by sjplimp — APPROVED** [review]

File: doc/src/Tools.rst [hunk 1]
```diff
@@ -376,21 +376,36 @@ See README file in the tools/fep directory.
...
-i-pi tool
+i-PI tool
 -------------------
 
-The tools/i-pi directory contains a version of the i-PI package, with
-all the LAMMPS-unrelated files removed.  It is provided so that it can
-be used with the :doc:`fix ipi <fix_ipi>` command to perform
-path-integral molecular dynamics (PIMD).
+The tools/i-pi directory used to contain a version of the i-PI package.
+This version, however, was not updated and thus have become outdated.
```
Comment (stanmoore1) [inline doc/src/Tools.rst]: I think "was not updated and thus have become outdated" is a little confusing.  Maybe it should say something like:

The tools/i-pi directory used to contain a version of the i-PI package, however, this version was removed and i-PI is now available via PyPi using the pip package manager at ...

File: tools/i-pi/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+This folder used to contain a bundled version of [i-PI](https://ipi-code.org).
+However, the bundled version was not updated and thus had become outdated.
```
Comment (stanmoore1) [inline tools/i-pi/README.md]: However, the bundled version was removed and i-PI is now available via PyPi ...

File: doc/src/Tools.rst (hunk 1 above)
Comment (akohlmey) [reply doc/src/Tools.rst]: Done

File: tools/i-pi/README.md (hunk 2 above)
Comment (akohlmey) [reply tools/i-pi/README.md]: Done

File: doc/src/Commands_removed.rst [hunk 3]
```diff
@@ -148,6 +148,14 @@ performance characteristics on NVIDIA GPUs. Both, the KOKKOS
...
 and allow running LAMMPS with GPU acceleration.
 
+i-PI tool
+---------
+
+.. deprecated:: TBD
+
+The i-PI tool has been removed from the LAMMPS distribution.  Instead,
+instructions to install i-PI from PyPi via pip are provided.
+
```
Comment (ceriottm) [inline doc/src/Commands_removed.rst:158]: If possible I wouldn't indicate it is "deprecated" as we are simply pointing users to a "better" way to access it. I'd say something to the effect of 

```
The i-PI tool is no longer bundled with the LAMMPS package. Users are encouraged to 
install the most recent version from PyPI. Instructions to do so using `pip` are provided.

File: doc/src/Tools.rst [hunk 4]
```diff
@@ -376,21 +376,40 @@ See README file in the tools/fep directory.
...
 
 The i-PI package was created and is maintained by Michele Ceriotti,
 michele.ceriotti at gmail.com, to interface to a variety of molecular
 dynamics codes.
 
-See the tools/i-pi/manual.pdf file for an overview of i-PI, and the
-:doc:`fix ipi <fix_ipi>` page for further details on running PIMD
-calculations with LAMMPS.
+i-PI is now available via PyPi using the pip package manager at:
+https://pypi.org/project/i-PI/
```
Comment (ceriottm) [inline doc/src/Tools.rst]: Actually we moved renamed the main PyPI project to `ipi` (which is the name of the import). Still figuring out how to deprecate the old one. The "right" link is thus
`https://pypi.org/project/ipi/`

File: doc/src/Tools.rst [hunk 5]
```diff
@@ -376,21 +376,40 @@ See README file in the tools/fep directory.
...
+
+Here are the commands to set up a virtual environment and install
+i-PI into it with all its dependencies.
+
+.. code-block:: sh
+
+   python -m venv ipienv
+   source ipienv/bin/activate
+   pip install --upgrade pip
+   pip install i-PI
```
Comment (ceriottm) [inline doc/src/Tools.rst]: pip install ipi

File: doc/src/fix_ipi.rst
Comment (ceriottm) [inline doc/src/fix_ipi.rst:1]: Could you please merge with master so I can suggest edits on the latest version?

File: tools/i-pi/README.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+This folder used to contain a bundled version of [i-PI](https://ipi-code.org)
+but the bundled version was removed in 05/2024.
```
Comment (ceriottm) [inline tools/i-pi/README.md]: likely 06/2024 to be pedantic ^_^

File: tools/i-pi/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+i-PI is available via PyPi using the pip package manager at:
+https://pypi.org/project/i-PI/
```
Comment (ceriottm) [inline tools/i-pi/README.md]: https://pypi.org/project/ipi

File: tools/i-pi/README.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+Here are the commands to set up a virtual environment and
+install i-PI into it with all its dependencies.
+
+``` sh
+python -m venv ipienv
+source ipienv/bin/activate
+pip install --upgrade pip
+pip install i-PI
```
Comment (ceriottm) [inline tools/i-pi/README.md]: pip install ipi

**Review by ceriottm — CHANGES_REQUESTED** [review]
I would do the changes myself but I think I'd have to get write permission to your fork. You tell me how you'd rather do this. I think this is all I have to comment on, except probably for the documentation in fix_ipi.rst that I'd first synch up with that in `develop`. Thanks for your help, as usual!

File: doc/src/Commands_removed.rst (hunk 3 above)
Comment (akohlmey) [reply doc/src/Commands_removed.rst:158]: I changed this to ".. versionchanged::"

File: doc/src/Tools.rst (hunk 4 above)
Comment (akohlmey) [reply doc/src/Tools.rst]: done

File: doc/src/Tools.rst (hunk 5 above)
Comment (akohlmey) [reply doc/src/Tools.rst]: done

**Comment by akohlmey** [comment]
> I would do the changes myself but I think I'd have to get write permission to your fork.

You would have to become a core LAMMPS developer with write access to the lammps repositories. That would give you write access to all pull request branches, except when the user has disabled this feature (I didn't) on submission of the pull request, or when the pull request is created from an institutional account (mine is private). For one of the two reasons, the LAMMPS developers cannot push changes into your fork directly.

> You tell me how you'd rather do this. I think this is all I have to comment on, except probably for the documentation in fix_ipi.rst that I'd first synch up with that in `develop`. Thanks for your help, as usual!

When you comment on individual changes there, there is a `[+/-]` icon in the editor (to the right of Preview), where you can add a suggested change for the referenced line. I can just click on that suggestion and github will add this as a commit to my branch.

Otherwise you can fetch my repository and check out my branch, make/commit changes, and then submit this as a pull request. By default, this will submit a pull request to the LAMMPS main repo, but you can just change `lammps/lammps` to `akohlmey/lammps` in the comparison URL and it will then be set up to file a pull request to *my* fork. Once I merge that, it will update the pull request automatically (since it is the same branch).

File: tools/i-pi/README.md (hunk 8 above)
Comment (akohlmey) [reply tools/i-pi/README.md]: done

File: tools/i-pi/README.md (hunk 7 above)
Comment (akohlmey) [reply tools/i-pi/README.md]: done

File: tools/i-pi/README.md (hunk 6 above)
Comment (akohlmey) [reply tools/i-pi/README.md]: done

File: doc/src/fix_ipi.rst
Comment (akohlmey) [reply doc/src/fix_ipi.rst:1]: done

File: doc/src/Commands_removed.rst (hunk 3 above)
Comment (ceriottm) [reply doc/src/Commands_removed.rst:158]: Good with me.

**Review by ceriottm — APPROVED** [review]
Actually it's all fine with me. Thanks a lot, can be merged whenever you want.

File: tools/i-pi/README.md (hunk 6 above)
Comment (stanmoore1) [reply tools/i-pi/README.md]: Does this mean I can't merge this PR until next week? 😆

Comment (ceriottm) [reply tools/i-pi/README.md]: You people are too efficient, I clearly underestimated you ^_^'

Comment (akohlmey) [reply tools/i-pi/README.md]: @stanmoore1 it also means that you have to do a feature release within the next month. :wink:

**Comment by akohlmey** [comment]
@ceriottm Now that the old i-PI distribution has been removed from LAMMPS, you may also what to check and update the examples for fix ipi in the LAMMPS distribution and possibly update them. Please see this post in the LAMMPS forum: https://matsci.org/t/how-to-fix-i-ip-nameerror-tag-name-initialize-is-not-a-recognized-property-of-simulation-objects/55725

**Comment by ceriottm** [comment]
> @ceriottm Now that the old i-PI distribution has been removed from LAMMPS, you may also what to check and update the examples for fix ipi in the LAMMPS distribution and possibly update them. Please see this post in the LAMMPS forum: https://matsci.org/t/how-to-fix-i-ip-nameerror-tag-name-initialize-is-not-a-recognized-property-of-simulation-objects/55725

OOOPS! That was fast. Sure, looking into this.

**Pull request merged**